cdap-api-common/src/main/java/co/cask/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -108,30 +108,78 @@ private Schema read(JsonReader reader, Map<String, Schema> knownRecords) throws
         return readUnion(reader, knownRecords);
       case BEGIN_OBJECT: {
         reader.beginObject();
-        String name = reader.nextName();
-        if (!"type".equals(name)) {
-          throw new IOException("Property \"type\" missing.");
+        // Type of the schema
```

Comment:
this method seems way too long now. any way to break it up? can the reading of an object move to a different method?